test/unit/reductions/exponentiated_gradient/test_exponentiatedgradient_smoke.py

```diff
@@ -10,268 +11,335 @@
          "n_predictors": 4},
-        {"cons_class": ErrorRateParity, "eps": 0.01,
+        {"constraint_class": ErrorRateParity, "eps": 0.01,
          "best_gap": 0.000000, "last_iter": 5,
          "best_iter": 5, "disp": 0.010000,
          "error": 0.272916, "n_oracle_calls": 26, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 5, "ratio": 0.8},
-        {"cons_class": ErrorRateParity, "eps": 0.005,
+        {"constraint_class": ErrorRateParity, "eps": 0.005,
          "best_gap": 0.000000, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.329444, "n_oracle_calls": 19, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 5},
-        {"cons_class": ErrorRateParity, "eps": 0.005,
+        {"constraint_class": ErrorRateParity, "eps": 0.005,
          "best_gap": 0.000000, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.275347, "n_oracle_calls": 26, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 5, "ratio": 0.8},
         # ================================================
-        {"cons_class": TruePositiveRateParity, "eps": 0.005,
+        {"constraint_class": TruePositiveRateParity, "eps": 0.005,
          "best_gap": 0.0, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.25, "n_oracle_calls": 16, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 2},
-        {"cons_class": FalsePositiveRateParity, "eps": 0.005,
+        {"constraint_class": FalsePositiveRateParity, "eps": 0.005,
          "best_gap": 0.0, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.427133, "n_oracle_calls": 17, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 3}
     ]
     smoke_test_data_flipped = [
-        {"cons_class": TruePositiveRateParity, "eps": 0.005,
+        {"constraint_class": TruePositiveRateParity, "eps": 0.005,
          "best_gap": 0.0, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.427133, "n_oracle_calls": 17, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 3},
-        {"cons_class": FalsePositiveRateParity, "eps": 0.005,
+        {"constraint_class": FalsePositiveRateParity, "eps": 0.005,
          "best_gap": 0.0, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.25, "n_oracle_calls": 16, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 2},
-        {"cons_class": EqualizedOdds, "eps": 0.005,
+        {"constraint_class": EqualizedOdds, "eps": 0.005,
          "best_gap": 0.000000, "last_iter": 5,
          "best_iter": 5, "disp": 0.005000,
          "error": 0.442883, "n_oracle_calls": 19, "n_oracle_calls_dummy_returned": 0,
          "n_predictors": 6}
     ]
 
-    def run_smoke_test(self, data, flipped=False):
-        if flipped:
-            y = self.flipped_y
-        else:
-            y = self.y
+    smoke_test_data_regression = [
```

Comment:
Two issues with the current tests:
* you also need to check `expgrad.weights_` not just `expgrad.predictors_`
* it would be good to try setting `upper_bound` to a value that's different from `eps` and something that would make the problem feasible--currently all your examples have infeasible constraints... so you could maybe try `upper_bound=0.05`, which would make some of the examples feasible and some infeasible; i'd also suggest setting `nu=1e-6` or `1e-3` and `max_iter=20` in these runs (since the automatic setting of `nu` needs to be double-checked and might change in the future)

Reply:
- yes! Updated.
- Good point. I thought that's an issue since the gap doesn't go to 0. Will add a few more tests, although I'll leave the existing ones there as well. In practice that can happen, too, and we should have tests covering that. I'm adding the tests for `upper_bound=0.05` with `best_gap=0.0`. Regarding `nu` and `max_iter` restrictions: The number of iterations is never larger than 5 or 6 so that wouldn't do anything, and that applies even when I set `nu` to 1e-6. I've added it anyway for now, but just so you're aware.